### PR TITLE
Standardized request options for all iterators

### LINKED ISSUE #463: Iterator request options need to be standardized
The iterator request options are not currently the same across the different types. New request options should be added to remove unnecessary options and updated to match across the SDK.

### PR DESCRIPTION
**Pull request by j82w**
Pull Request Template

## Description

1. Added IteratorRequestOptions removes all query only options
2. Added ItemIteratorRequestOptions which extends IteratorRequestOptions. This adds PartitionKey support for read feed on item read feed.
3. QueryRequestOptions now extends ItemIteratorRequestOptions.
4. Updated all the iterator methods to use the new request options
## Type of change
5. Added database stream iterator

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Closing issues

Put closes #463

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Samples/CodeSamples/appSettings.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-  "EndPointUrl": "https://localhost:8081",
-  "AuthorizationKey": "Super secret key"
+  "EndPointUrl": "https://jawilleytemp.documents.azure.com:443/",
+  "AuthorizationKey": "GboKIMLEDVsipUwUx9qVo9k7zACZqhOG3Kh9PkWPeZK2FuSAAEeG6R3T3TinPKh3upxV4OYY4uw6tJI7whtfDA=="
```
Comment (bartelink) [inline Microsoft.Azure.Cosmos.Samples/CodeSamples/appSettings.json]: Er...

Comment (j82w) [reply Microsoft.Azure.Cosmos.Samples/CodeSamples/appSettings.json]: Thanks, the key has been changed.

File: Microsoft.Azure.Cosmos/src/CosmosClient.cs [hunk 2]
```diff
@@ -557,9 +598,8 @@ internal DatabaseProperties PrepareDatabaseProperties(string id)
         {
             Debug.Assert(state == null);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/CosmosClient.cs]: How about typed based on stream?

File: Microsoft.Azure.Cosmos/src/RequestOptions/FeedRequestOptions.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+//------------------------------------------------------------
+
+namespace Microsoft.Azure.Cosmos
+{
+    using System.Globalization;
+    using Microsoft.Azure.Documents;
+    using static Microsoft.Azure.Documents.RuntimeConstants;
+
+    /// <summary>
+    /// The Cosmos query request options
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/RequestOptions/FeedRequestOptions.cs]: query in description

**Comment by j82w** [comment]
This PR is getting closed in favor of merging read feed and query into a single API since read feed is just select * query.

**Comment by j82w** [comment]
This PR is getting closed in favor of merging read feed and query into a single API since read feed is just select * query.

**Pull request closed without merging**